MediaAdmin/Tests/Security/Authorization/Voter/MediaFolderGroupRoleVoterTest.php

```diff
@@ -0,0 +1,163 @@
+     * @param bool   $supports
+     *
+     * @dataProvider provideAttributeAndSupport
+     */
+    public function testSupportsAttribute($attribute, $supports)
+    {
+        $this->assertSame($supports, $this->voter->supportsAttribute($attribute));
+    }
+
+    /**
+     * @return array
+     */
+    public function provideAttributeAndSupport()
+    {
+        return array(
+            array('ROLE_ACCESS_TREE_GENERAL_NODE', false),
+            array('ROLE_ACCESS_REDIRECTION', false),
+            array('ROLE_ACCESS_MEDIA_FOLDER', true),
+            array('ROLE_ACCESS_CREATE_MEDIA_FOLDER', true),
+            array('ROLE_ACCESS_UPDATE_MEDIA_FOLDER', true),
+            array('ROLE_ACCESS_DELETE_MEDIA_FOLDER', true),
+            array('ROLE_ACCESS_CREATE_MEDIA', true),
+            array('ROLE_ACCESS_UPDATE_MEDIA', true),
+            array('ROLE_ACCESS_DELETE_MEDIA', true),
+            array('ROLE_ADMIN', false),
+            array('ROLE_USER', false),
+            array('ROLE_FROM_PUBLISHED_TO_DRAFT', false),
+        );
+    }
+
+    /**
+     * @param int     $expectedVoterResponse
+     * @param string  $folderId
+     * @param string  $mfgrFolderId
+     * @param string  $mfgrRole
+     * @param boolean $isGranted
+     * @param string  $groupSiteId
+     *
+     * @dataProvider provideResponseAndNodeData
+     */
+    public function testVote($expectedVoterResponse, $folderId, $mfgrFolderId, $mfgrRole, $isGranted, $groupSiteId = 'siteId')
+    {
+        $siteId = 'siteId';
+        $role = TreeFolderPanelStrategy::ROLE_ACCESS_UPDATE_MEDIA_FOLDER;
+        $folder = Phake::mock('OpenOrchestra\MediaModelBundle\Document\Folder');
+        Phake::when($folder)->getId()->thenReturn($folderId);
+        Phake::when($folder)->hasSite($siteId)->thenReturn(true);
+
+        $mediaFolderGroupRole = Phake::mock('OpenOrchestra\Media\Model\MediaFolderGroupRoleInterface');
+        Phake::when($mediaFolderGroupRole)->isGranted()->thenReturn($isGranted);
+
+        $group = $this->generateGroup($groupSiteId);
+        Phake::when($group)->getMediaFolderRoleByMediaFolderAndRole($mfgrFolderId, $mfgrRole)->thenReturn($mediaFolderGroupRole);
+        $otherGroup = $this->generateGroup('otherSiteId');
+        $noSiteGroup = $this->generateGroup();
+
+        $user = Phake::mock('OpenOrchestra\UserBundle\Model\UserInterface');
+        Phake::when($user)->getGroups()->thenReturn(array($noSiteGroup, $otherGroup, $group));
+        $token = Phake::mock('Symfony\Component\Security\Core\Authentication\Token\TokenInterface');
+        Phake::when($token)->getUser()->thenReturn($user);
+        $this->assertSame($expectedVoterResponse, $this->voter->vote($token, $folder, array($role)));
+    }
+
+    /**
+     * @return array
+     */
+    public function provideResponseAndNodeData()
+    {
+        return array(
+            array(VoterInterface::ACCESS_GRANTED, 'folderId', 'folderId', TreeFolderPanelStrategy::ROLE_ACCESS_UPDATE_MEDIA_FOLDER, true),
```

Comment:
small tip : you could add a string to summarize the test case as the key of the value in the array